### PR TITLE
ww-code-test

### PR DESCRIPTION
**Pull request by derickdss**
Hi,

Please review my code to complete the test consisting of PArt A,B ancd C

Part A:
This appeared a bit challenging to start with but thanks to the clue of where the problem lies, I was able to get the solution after that. I wasn't too well versed with Rambda and did some improvisations like `R.subtract(a,b) === 0` instead of `a === b`.
In the end, I did the changes to get all test to pass.

Part B:
This was  a bit challengin and in my opnion, I haven't really solved the problem inspite of getting the test to pass. I saw that the NI was being calculated for the wrong year as rundate wasn't being passed an argument. This caused the backend to use some default which caused an incorrect mapping to get the floor, ceiling values need by the slice function. As a solution, I inserted the current tax year in the failing tests of 2018-19 in the feature file

Part C:
I have created a very basic react frontend page for the user to enter his income and the NI results are dynamically calculated as the user enters his inputs. I have thought on initially using a form(hence Materialui) to get the user to enter his inputs but decided against it for easier usage and appearance to the user. 
However, on basic ibservation, I saw the same issue where the correct NI band mapping wasn't happening due to the default rundate value. I tried to debug this took too many wrong turns and therefore decided to insert a cheat fix in the frontend just to show you that I had a clue of the problem but couldnt quite trace it in the backend.

In conclusion, it was a nice test, but a bit of unfamiliar territory with the rambda bits.

Note: I wasn't able to invite rob1256

### CODE REVIEW

**Comment by mrwithersea** [comment]
@derickdss, thanks for submitting your code test, we will aim to review this and feedback to you in the next 24 hours.

Many thanks
Marc
